sys/vm/vm_pageout.c

```diff
@@ -2362,23 +2362,21 @@ vm_pageout(void)
 {
 	struct proc *p;
 	struct thread *td;
-	int error, first, i, j, pageout_threads;
+	int error, i, j, pageout_threads;
 
 	p = curproc;
 	td = curthread;
 
 	mtx_init(&vm_oom_ratelim_mtx, "vmoomr", NULL, MTX_DEF);
 	swap_pager_swap_init();
-	for (first = -1, i = 0; i < vm_ndomains; i++) {
+	for (i = 0; i < vm_ndomains; i++) {
 		if (VM_DOMAIN_EMPTY(i)) {
 			if (bootverbose)
 				printf("domain %d empty; skipping pageout\n",
 				    i);
 			continue;
 		}
-		if (first == -1)
-			first = i;
-		else {
+		if (i != 0) {
 			error = kthread_add(vm_pageout_worker,
```

Comment:
I don't think this is correct at all.
(a) first will be != 0 if VM_DOMAIN_EMPTY(0) is true.

Reply:
You are right. I missed considering the case when vm domain 0 is empty.